messaging-client-core/src/main/java/com/inmobi/messaging/consumer/MessageConsumer.java

```diff
@@ -83,4 +83,13 @@ public Message next(long timeout, TimeUnit timeunit)
    * Close and cleanup all resources such as connection, file handles etc.
    */
   public void close();
+
+  /***
+   * Calculates the remaining amount of data for a consumer. Other consumer implementations are
```

Comment:
Rephrase the doc to: "Returns the amount of data in bytes present in stream that hasn't been read by consumer application yet." Remove the line "Other consumer..." and instead in throws section, mention ", UnSupportedException, if this method is not implemented".